lib/main.c

```diff
@@ -259,23 +271,177 @@ int surjectionproof_verify(const unsigned char *proof_data, const size_t proof_l
   secp256k1_context_destroy(ctx);
   return ret;
 }
 
-int ec_seckey_tweak_mul(unsigned char *key, const unsigned char *tweak) {
+int ec_seckey_tweak_mul(unsigned char *key, const unsigned char *tweak)
+{
   secp256k1_context *ctx = secp256k1_context_create(SECP256K1_CONTEXT_ALL);
   int ret = secp256k1_ec_seckey_tweak_mul(ctx, key, tweak);
   secp256k1_context_destroy(ctx);
   return ret;
 }
+
+int ec_is_valid_xonly_pubkey(const unsigned char *key)
+{
+  secp256k1_context *ctx = secp256k1_context_create(SECP256K1_CONTEXT_ALL);
+  secp256k1_xonly_pubkey pubkey;
+  int ret = secp256k1_xonly_pubkey_parse(ctx, &pubkey, key);
+  secp256k1_context_destroy(ctx);
+  return ret;
+}
+
+int ec_is_valid_pubkey(const unsigned char *key, size_t key_len)
+{
+  secp256k1_context *ctx = secp256k1_context_create(SECP256K1_CONTEXT_ALL);
+  secp256k1_pubkey pubkey;
+  int ret = secp256k1_ec_pubkey_parse(ctx, &pubkey, key, key_len);
+  secp256k1_context_destroy(ctx);
+  return ret;
+}
+
+int ec_is_point(const unsigned char *key, size_t key_len)
+{
+  if (key_len == 32) {
+    return ec_is_valid_xonly_pubkey(key);
+  }
+
+  return ec_is_valid_pubkey(key, key_len);
+}
+
+int ec_point_compress(unsigned char *output, size_t *output_len, const unsigned char *point, size_t point_len)
+{
+  secp256k1_context *ctx = secp256k1_context_create(SECP256K1_CONTEXT_ALL);
+  secp256k1_pubkey pubkey;
+  int ret = secp256k1_ec_pubkey_parse(ctx, &pubkey, point, point_len);
+  if (ret == 1)
+  {
+    ret = secp256k1_ec_pubkey_serialize(ctx, output, output_len, &pubkey, SECP256K1_EC_COMPRESSED);
+  }
+  secp256k1_context_destroy(ctx);
+  return ret;
+}
+
+int ec_point_from_scalar(unsigned char *output, size_t *output_len, const unsigned char *scalar)
+{
+  secp256k1_context *ctx = secp256k1_context_create(SECP256K1_CONTEXT_ALL);
+  secp256k1_pubkey pubkey;
+  int ret = secp256k1_ec_pubkey_create(ctx, &pubkey, scalar);
+  if (ret == 1)
+  {
+    ret = secp256k1_ec_pubkey_serialize(ctx, output, output_len, &pubkey, SECP256K1_EC_COMPRESSED);
+  }
+  secp256k1_context_destroy(ctx);
+  return ret;
+}
+
+int ec_x_only_point_add_tweak(unsigned char *output, int* parity, const unsigned char *point, const unsigned char *tweak)
+{
+  secp256k1_context *ctx = secp256k1_context_create(SECP256K1_CONTEXT_ALL);
+  secp256k1_xonly_pubkey pubkey;
+  secp256k1_pubkey pubkey_result;
+  int ret = secp256k1_xonly_pubkey_parse(ctx, &pubkey, point);
+  if (ret == 1)
+  {
+    ret = secp256k1_xonly_pubkey_tweak_add(ctx, &pubkey_result, &pubkey, tweak);
+    if (ret == 1)
+    {
+      ret = secp256k1_xonly_pubkey_from_pubkey(ctx, &pubkey, parity, &pubkey_result);
+      if (ret == 1)
+      {
+        ret = secp256k1_xonly_pubkey_serialize(ctx, output, &pubkey);
+      }
+    }
+  }
+  secp256k1_context_destroy(ctx);
+  return ret;
+}
+
+int ec_ecdsa_sign(unsigned char *output, const unsigned char *d, const unsigned char *h, int withextradata, const unsigned char *e)
```

Comment:
there's some inconsistency between these sign/verify methods and those for schnorr. I would rename ECDSA methods to `ec_sign_ecsa` and `ec_verify_ecdsa`.